### PR TITLE
New tiles design

### PR DESCRIPTION
**Pull request by stonebuzz**
### Changes description

Add new tilmes design

Keep the standard and rename it as 'legacy'
Add new design -> fix

Legacy :
image

Fix : 
image

Add capability to expand tile if needed (depend of description + title length) with ```See more```

image

### Checklist

Please check if your PR fulfills the following specifications:

- [ ] Tests for the changes have been added
- [ ] Docs have been added/updated

### References

Closes #N/A

### CODE REVIEW

File: css/styles.scss [hunk 1]
```diff
@@ -668,6 +668,11 @@ tr[data-itemtype="PluginFormcreatorCondition"] ~ tr[data-itemtype="PluginFormcre
    background-color: #CCC;
 }
 
+.plugin_formcreator_formTile.tile_template_fix {
```
Comment (btry) [inline css/styles.scss:671]: tile_template_fix should not mention the word "fix"

File: inc/entityconfig.class.php [hunk 2]
```diff
@@ -66,6 +66,10 @@ class PluginFormcreatorEntityconfig extends CommonDBTM {
    const CONFIG_SEARCH_ISSUE_HIDDEN = 0;
    const CONFIG_SEARCH_ISSUE_VISIBLE = 1;
 
+   const CONFIG_UI_FORM_LEGACY = 'legacy';
+   const CONFIG_UI_FORM_FIX = 'fix';
```
Comment (btry) [inline inc/entityconfig.class.php:70]: Some here. Why not something like "uniform_height" ? "V_aligned"

File: install/install.php [hunk 3]
```diff
@@ -263,7 +263,7 @@ protected function configureExistingEntities() {
 
       /** Value -2 is "inheritance from parent" @see PluginFormcreatorEntityconfig::CONFIG_PARENT */
       $query = "INSERT INTO glpi_plugin_formcreator_entityconfigs
-                  (entities_id, replace_helpdesk, default_form_list_mode, sort_order, is_kb_separated, is_search_visible, is_dashboard_visible, is_header_visible, is_search_issue_visible)
+                  (entities_id, replace_helpdesk, default_form_list_mode, sort_order, is_kb_separated, is_search_visible, is_dashboard_visible, is_header_visible, is_search_issue_visible, form_design)
```
Comment (btry) [inline install/install.php:266]: Tile_design instead, as this is not related to form themselves, and will also apply on kB items.

**Review by btry — CHANGES_REQUESTED** [review]

File: install/install.php (hunk 3 above)
Comment (stonebuzz) [reply install/install.php]: done

File: inc/entityconfig.class.php (hunk 2 above)
Comment (stonebuzz) [reply inc/entityconfig.class.php]: done

File: css/styles.scss (hunk 1 above)
Comment (stonebuzz) [reply css/styles.scss]: done

File: inc/entityconfig.class.php [hunk 4]
```diff
@@ -146,6 +150,14 @@ public static function getEnumSearchIssueVisibility() : array {
       ];
    }
 
+   public static function getEnumUIForm() : array {
+      return [
+         self::CONFIG_PARENT                 => __('Inheritance of the parent entity'),
+         self::CONFIG_UI_FORM_MASONRY        => __('Standart', 'formcreator'),
```
Comment (btry) [inline inc/entityconfig.class.php:156]:
```suggestion
         self::CONFIG_UI_FORM_MASONRY        => __('Variable height', 'formcreator'),
```

File: inc/entityconfig.class.php [hunk 5]
```diff
@@ -146,6 +150,14 @@ public static function getEnumSearchIssueVisibility() : array {
...
+         self::CONFIG_UI_FORM_UNIFORM_HEIGHT => __('Fix tile', 'formcreator'),
```
Comment (btry) [inline inc/entityconfig.class.php:157]:
```suggestion
         self::CONFIG_UI_FORM_UNIFORM_HEIGHT => __('Uniform height', 'formcreator'),
```

File: inc/form.class.php [hunk 6]
```diff
@@ -690,6 +690,11 @@ public function showWizard() : void {
    public static function getFormList(int $rootCategory = 0, string $keywords = '', bool $helpdeskHome = false): array {
       global $DB, $TRANSLATE;
 
+      $display_format = PluginFormcreatorEntityconfig::CONFIG_UI_FORM_MASONRY;
+      if (PluginFormcreatorEntityconfig::getUsedConfig('tile_design', Session::getActiveEntity()) == PluginFormcreatorEntityconfig::CONFIG_UI_FORM_UNIFORM_HEIGHT) {
+         $display_format = PluginFormcreatorEntityconfig::CONFIG_UI_FORM_UNIFORM_HEIGHT;
+      }
```
Comment (btry) [inline inc/form.class.php:696]: Why not simply write the following ? If the entity configrow does not exists, it is created by getUsedConfig().

```suggestion
      $display_format = PluginFormcreatorEntityconfig::getUsedConfig('tile_design', Session::getActiveEntity());
```

File: inc/form.class.php [hunk 7]
```diff
@@ -821,7 +827,8 @@ public static function getFormList(int $rootCategory = 0, string $keywords = '',
                'description'      => '',
                'type'             => 'faq',
                'usage_count'      => $faq['view'],
-               'is_default'       => false
+               'is_default'       => false,
+               'tile_template'         => 'fix'
```
Comment (btry) [inline inc/form.class.php:831]:
```suggestion
               'tile_template'    => 'fix'
```

The value 'fix' does not matches any of your new constants.

File: inc/entityconfig.class.php [hunk 8]
```diff
@@ -66,6 +66,10 @@ class PluginFormcreatorEntityconfig extends CommonDBTM {
    const CONFIG_SEARCH_ISSUE_HIDDEN = 0;
    const CONFIG_SEARCH_ISSUE_VISIBLE = 1;
 
+   const CONFIG_UI_FORM_MASONRY = 'masonry';
+   const CONFIG_UI_FORM_UNIFORM_HEIGHT = 'uniform_height';
```
Comment (btry) [inline inc/entityconfig.class.php:70]: Other settings use numbers

```suggestion
   const CONFIG_UI_FORM_MASONRY = 0;
   const CONFIG_UI_FORM_UNIFORM_HEIGHT = 1;
```

File: install/install.php [hunk 9]
```diff
@@ -272,7 +272,8 @@ protected function configureExistingEntities() {
                   IF(ent.id = 0, 0, -2),
                   IF(ent.id = 0, 1, -2),
                   IF(ent.id = 0, 0, -2),
-                  IF(ent.id = 0, 1, -2)
+                  IF(ent.id = 0, 1, -2),
+                  'legacy'
```
Comment (btry) [inline install/install.php:276]: The setting must be inheritable from parent entity. Also the  value **legacy** does not matches any constant

The following change uses -2 for the inheritance, like other settings. The value 0 means 'variable height', which is the legacy setting. 

```suggestion
                  IF(ent.id = 0, 0, -2),
```

If using the uniform heigh setting by default is wanted, then let's use the suggestion  below
```suggestion
                  IF(ent.id = 0, 1, -2),
```

File: install/mysql/plugin_formcreator_empty.sql [hunk 10]
```diff
@@ -38,6 +38,7 @@ CREATE TABLE IF NOT EXISTS `glpi_plugin_formcreator_entityconfigs` (
   `is_dashboard_visible`    int(11) NOT NULL DEFAULT '-2',
   `is_header_visible`       int(11) NOT NULL DEFAULT '-2',
   `is_search_issue_visible` int(11) NOT NULL DEFAULT '-2',
+  `tile_design`             varchar(255) NOT NULL DEFAULT 'legacy',
```
Comment (btry) [inline install/mysql/plugin_formcreator_empty.sql:41]:
```suggestion
  `tile_design`             int(11) NOT NULL DEFAULT '-2',
```

File: install/upgrade_to_2.13.php [hunk 11]
```diff
@@ -59,6 +59,11 @@ public function addEntityOption() {
          $this->migration->addField($table, 'is_search_issue_visible', 'integer', ['after' => 'is_header_visible', 'value' => '-2']);
          $this->migration->addPostQuery("UPDATE `glpi_plugin_formcreator_entityconfigs` SET `is_search_issue_visible`= 1 WHERE `entities_id` = 0");
       }
+
+      if (!$DB->fieldExists($table, 'tile_design')) {
+         $this->migration->addField($table, 'tile_design', 'string', ['after' => 'is_search_issue_visible', 'value' => 'legacy']);
+         $this->migration->addPostQuery("UPDATE `glpi_plugin_formcreator_entityconfigs` SET `tile_design`= 'legacy' WHERE `entities_id` = 0");
+      }
```
Comment (btry) [inline install/upgrade_to_2.13.php:66]:
```suggestion
      if (!$DB->fieldExists($table, 'tile_design')) {
         $this->migration->addField($table, 'tile_design', 'string', ['after' => 'is_search_issue_visible', 'value' => '-2']);
         $this->migration->addPostQuery("UPDATE `glpi_plugin_formcreator_entityconfigs` SET `tile_design`= '0' WHERE `entities_id` = 0");
      }
```

File: js/scripts.js [hunk 12]
```diff
@@ -370,9 +386,15 @@ function buildTiles(list) {
             url = rootDoc + '/front/knowbaseitem.form.php?id=' + item.id;
          }
 
+         var tiles_design = "";
+         console.log(item.tile_template);
```
Comment (btry) [inline js/scripts.js:390]: I guess this is for debug purpose

File: js/scripts.js [hunk 13]
```diff
@@ -370,9 +386,15 @@ function buildTiles(list) {
...
+         if(item.tile_template == "uniform_height"){
+            tiles_design = "tile_design_uniform_height";
+         }
```
Comment (btry) [inline js/scripts.js:393]:
```suggestion
         if (item.tile_template == "1") { // @see PluginFormcreatorEntityConfig::CONFIG_UI_FORM_UNIFORM_HEIGHT
            tiles_design = "tile_design_uniform_height";
         }
```

**Review by btry — CHANGES_REQUESTED** [review]

File: inc/form.class.php [hunk 14]
```diff
@@ -821,7 +824,8 @@ public static function getFormList(int $rootCategory = 0, string $keywords = '',
                'description'      => '',
                'type'             => 'faq',
                'usage_count'      => $faq['view'],
-               'is_default'       => false
+               'is_default'       => false,
+               'tile_template'    => 'fix'
```
Comment (btry) [inline inc/form.class.php:828]: I think I missed this in the previuous review.

```suggestion
               'tile_template'    => PluginFormcreatorEntityConfig::CONFIG_UI_FORM_MASONRY
```

**Review by btry — CHANGES_REQUESTED** [review]

**Review by btry — APPROVED** [review]

**Comment by btry** [comment]
@stonebuzz : is this PR ready ? It looks good to me.
